### ISSUE TITLE
Allow aborting requests

### ISSUE THREAD

**Issue by pimterry** [apocas/dockerode#629]
For some commands, it would be useful to be able to abort them before they complete. I'm specifically thinking of `wait`, but I'm sure there's others too. There's been previous requests for this with image building, e.g. #506, #269

I think it should be easy to do so by accepting a `{ signal: AbortSignal }` option (like most promisified web APIs & modern Node.js APIs do) and passing that through directly to the HTTP requests, to become the `signal` option for [http.request](https://nodejs.org/docs/latest-v14.x/api/http.html#http_http_request_url_options_callback).

With that, everything should Just Work to allow API aborting easily everywhere. The end user code would look something like:

```javascript
const Docker = require('dockerode');
const docker = new Docker();

const abortController = new AbortController();
docker.getContainer('abc').wait({ signal: abortController.signal })

// Later:
abortController.abort(); // Ends the pending request
```

[AbortController](https://nodejs.org/api/globals.html#globals_class_abortcontroller) is supported in all v15 & v16 node versions, and in v14 behind a flag, and it's the standard way to abort all promisified APIs. Although that doesn't go back to node v8 (as Dockerode appears to support) there's no compatibility risk, since we wouldn't actually need to reference it here ourselves. We just need to pass through the `signal` property to the HTTP APIs. Any user using v15/v16 version, or using v14 with either the flag enabled or their own polyfill would be able to use it immediately. Anybody on older unsupported versions wouldn't have a problem, they just can't use it without a modern Node version.

Would you be open to that?

**Comment by apocas** [comment]
Sure, I think it's a good improvement. Without breaking changes is even better :)

**Comment by pimterry** [comment]
Ok, great. I'll look into this and open a PR soon.

**Comment by pimterry** [comment]
Hi @apocas, if you get a minute, is it possible to release the docker-modem change above that was merged & review the dockerode PR here (#630)? I'd love to be able to start using this! Let me know if there's anything at all that I can do to help.
